File: src/game/entity/enemies/npcs/Zombie.java
package game.entity.enemies.npcs;

import game.entity.models.Coordinates;
import game.utils.Paths;

public class Zombie extends IntelligentEnemy {
    public Zombie() {
        super();
        setMaxHp(300);
        setHp(getMaxHp());
    }

    public Zombie(Coordinates coordinates) {
        super(coordinates);
        setMaxHp(300);
        setHp(getMaxHp());
    }
    @Override
    public float getSpeed() {
        return 0.5f;
    }

    @Override
    protected String getBasePath() {
        return Paths.getEnemiesFolder() + "/zombie";
    }

    @Override
    public String[] getCharacterOrientedImages() {
        return new String[] {
                String.format("%s/zombie_%s_0.png", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/zombie_%s_1.png", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/zombie_%s_2.png", getBasePath(), imageDirection.toString().toLowerCase()),
                String.format("%s/zombie_%s_3.png", getBasePath(), imageDirection.toString().toLowerCase())
        };
    }
}

File: src/game/entity/enemies/npcs/Zombie.kt
package game.entity.enemies.npcs

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.utils.Paths.enemiesFolder

class Zombie : IntelligentEnemy {
    constructor(id: Long) : this() {
        this.id = id
    }
    constructor() : super()
    constructor(coordinates: Coordinates?) : super(coordinates) {}

    override fun getSpeed(): Float = 0.5f

    override fun getMaxHp(): Int = 300

    override val entitiesAssetsPath: String get() ="$enemiesFolder/zombie"

    override fun getCharacterOrientedImages(): Array<String> = Array(4) {
        "$entitiesAssetsPath/zombie_${imageDirection.toString().lowercase()}_$it.png"
    }

    override val type: EntityTypes
        get() = EntityTypes.Zombie
}